test/integration/keep-alive-with-relay.js

```diff
@@ -0,0 +1,140 @@
+const test = require('brittle')
+const { spawnFixture } = require('../helpers')
+const { Server: RelayServer } = require('blind-relay')
+const DHT = require('../../')
+const path = require('path')
+
+/*
+  When a peer connects to a relay server, there's two sockets established. One for data (between two peers)
+  and another "control" socket.
+  The relay server has a RELAY_KEEPALIVE that looks at the control socket to detect when a peer is no longer there.
+  The client and server both have a SOCKET_KEEPALIVE which detects when the other peer is no longer there.
+
+  A bug that occured with udx-native < 1.8.9 is that the relay server had detected the control socket had bad been destroyed,
+  but didn't fully clean up internally. That meant that this information was never fully relayed to the other peer.
+*/
+test('When Server is killed, Client should detect this - through relay', async t => {
+  t.plan(3)
+
+  const relayTest = t.test('relay')
+  relayTest.plan(3)
+  const clientTest = t.test('client')
+  clientTest.plan(3)
+  const serverTest = t.test('server')
+  serverTest.plan(3)
+
+  const RELAY_KEEPALIVE = 500
+  const SOCKET_KEEPALIVE = 10 * RELAY_KEEPALIVE
+
+  const clientKeyPair = DHT.keyPair()
+  const clientNode = new DHT()
+  const relayNode = new DHT()
+  const relayKeyPair = DHT.keyPair()
+  const serverKeyPair = DHT.keyPair()
+  const serverPublicKey = serverKeyPair.publicKey.toString('hex')
+  const serverSecretKey = serverKeyPair.secretKey.toString('hex')
+  let hasClientDetectedThatServerDied = false
+  let didClientNotDetectThatServerDiedTimer
+  let didClientNotDetectThatServerDiedTimerFired = false
+
+  t.teardown(async () => {
+    await clientNode.destroy()
+    await relayNode.destroy()
+  })
+
+  await startRelayServer()
+  startServer()
```

Comment:
async function, so I think it needs awaiting or a catch-handler to avoid uncaught exceptions